| 컬럼    | 역할    | 중요도   |
| ----- | ----- | ----- |
| pulse | 운동 강도 | ⭐⭐⭐⭐⭐ |
| time  | 운동 시간 | ⭐⭐⭐⭐  |
| kind  | 운동 종류 | ⭐⭐⭐⭐  |
| diet  | 식단    | ⭐⭐⭐   |
| id    | 식별자   | ❌     |


In [87]:
import seaborn as sns
import pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn.cluster import KMeans

# 데이터 로드
df = sns.load_dataset("exercise")

# 전처리
# 전처리
df["time"] = df["time"].map({"1 min": 1, "15 min": 15, "30 min": 30})
df["kind"] = df["kind"].map({
    "rest": 0,
    "walking": 1,
    "running": 2
})

# 🔥 핵심 추가
df["time"] = df["time"].astype(int)
df["kind"] = df["kind"].astype(int)

df = df.dropna()

# feature
X = df[["pulse", "time", "kind"]]

# 스케일링
scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

# 군집화
kmeans = KMeans(n_clusters=3, random_state=42)
df["Cluster"] = kmeans.fit_predict(X_scaled)

# 평균
print(df.groupby("Cluster")[["pulse", "time", "kind"]].mean())

# 분포 (핵심🔥)
print(pd.crosstab(df["Cluster"], df["kind"]))
print(pd.crosstab(df["Cluster"], df["time"]))

              pulse       time      kind
Cluster                                 
0         96.700000   5.666667  1.433333
1        124.941176  23.823529  2.000000
2         91.813953  18.720930  0.302326
kind      0   1   2
Cluster            
0         0  17  13
1         0   0  17
2        30  13   0
time     1   15  30
Cluster            
0        20  10   0
1         0   7  10
2        10  13  20


# 🏃‍♂️ 고객 세그먼테이션: 운동 데이터 군집화 분석

---

# 📊 1. 전체 구조 요약

* **Cluster 0** → 중간 강도 / 짧은 시간 / 걷기 + 달리기 혼합
* **Cluster 1** → 고강도 / 긴 시간 / 달리기
* **Cluster 2** → 낮은 강도 / 중간~긴 시간 / 휴식 + 일부 걷기

---

# 🧠 2. 군집별 상세 해석

---

## 🟢 Cluster 0

### 📊 평균

* pulse ≈ 96.7 (중간 강도)
* time ≈ 5.7 (짧은 시간)
* kind ≈ 1.43 (walking + running 섞임)

### 📊 분포

* kind → walking(17), running(13)
* time → 1분(20), 15분(10)

### 👉 해석

🚶‍♂️ **짧고 가볍게 운동하는 그룹**

* 짧게 운동
* 걷기 + 가벼운 러닝
* 강도는 높지 않음

👉 ✔ **라이트 운동층 / 바쁜 사람**

---

## 🔴 Cluster 1

### 📊 평균

* pulse ≈ 124.9 (매우 높은 강도🔥)
* time ≈ 23.8 (긴 시간)
* kind = 2 (100% running)

### 📊 분포

* kind → 전부 running (17)
* time → 30분(10), 15분(7)

### 👉 해석

🏃‍♂️ **고강도 운동 매니아**

* 달리기만 수행
* 심박수 매우 높음
* 운동 시간 길음

👉 ✔ **운동 매니아 / 선수형**

---

## 🟡 Cluster 2

### 📊 평균

* pulse ≈ 91.8 (낮은 강도)
* time ≈ 18.7 (중간~긴 시간)
* kind ≈ 0.3 (rest 중심)

### 📊 분포

* kind → rest(30), walking(13)
* time → 30분(20), 15분(13), 1분(10)

### 👉 해석

🛋️ **운동 거의 안 하는 그룹**

* 대부분 휴식 상태
* 가끔 걷기 수행
* 전반적으로 낮은 강도

👉 ✔ **비활동 / 건강관리 필요층**

---

# 🔥 3. 핵심 인사이트

* Cluster 0 → 가볍게 짧게 운동하는 사람
* Cluster 1 → 적극적으로 운동하는 사람
* Cluster 2 → 거의 운동하지 않는 사람

---

# 😲 4. 중요한 포인트

👉 우리가 **정답(label)을 주지 않았는데도**

```
운동 안함 / 적당히 / 열심히
```

👉 **자동으로 구분됨**

👉 👉 👉 이것이 바로 **군집화의 본질**

---

# 🧠 5. 비즈니스 관점 해석

## 💡 활용 전략

* **Cluster 1 (고강도)**
  → 프리미엄 운동 프로그램 추천
  → 고급 피트니스 서비스 타겟

* **Cluster 0 (중간층)**
  → 짧은 시간 운동 프로그램 추천
  → 직장인 맞춤 상품

* **Cluster 2 (비활동)**
  → 건강관리 / 운동 유도 캠페인
  → 입문용 운동 콘텐츠 제공

---

# 🚀 한 줄 정리

> 군집화는 데이터 속에서 사람의 행동 패턴을 발견하고, 그에 의미를 부여하는 과정이다.
